facet/README.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: pattern
+title: Facet
+folder: facet
+permalink: /patterns/facet/
+categories: Creational
+tags:
+ - Gang of Four
+---
+
+## Intent
+
+Use facet pattern to realize security, in order to ensure some privillage only can be granted to some certain users.
+
+## Explanation
+
+Real world example
+
+> Lets say if some client can only be able to read information of an object, then the client should be provided with a read-only facet and for write operation should be forbiddened.
+
+Wikipedia says
+
+> Facets are used as a security pattern in CapabilityOrientedProgramming, in order to satisfy the PrincipleOfLeastAuthority.
+
+**Programmatic Example**
+
+For a facet, use method to create, query the registered class and set the registered class.
+```java
+public abstract class Facet {
+  public static Facet create();
+  public static Class[] query(Facet f, Class[] interfaces);
+  public static Facet narrow(Facet f, Class[] interfaces);
+} 
+```
+For a sentry, use method to connect the facet and context to make request.
+```java
+public interface Sentry {
+  public abstract boolean execute(User user, Class interfaceClass);
+}
+```
+For the context, it abstracts the logic for make access control decisions on behalf of a Facet.
+```java
+public interface Context {
+  public boolean validateInterface(Class interfaceClass);
+}
+```
+After validation, execute the method.
+```java
+public interface SecurityMethods {
+  public static String delegate(User user);
+}
+```
+
+## Class Diagram
+
+![alt text](./etc/facet.urm.png "Facet pattern class diagram")
```

Comment:
Is this diagram made by the already installed plugin or have you generated it separately because this does not align with all the patterns that we already have here.

Reply:
e, I generate it separately, I suppose that the user call the facet to invoke the interface and the facet let the sentry to ask context whether the user request is valid. finally, if it is valid, then execute the request from user. For the above links, I cannot find the class diagram. Also, I cannot find some diagram from some other websites. So I generate it by myself. Is there any improvement I should do?

Reply:
I generate it based on the sequence diagram of the website: http://www.hillside.net/plop/plop2002/final/plop2002_ecrahen0_0.pdf
I make this pattern based on the information in this pdf.

Reply:
I will really appreciate it if you put the diagram that is generated by the URM plugin to eliminate any ambiguity in the types of images we have.

Reply:
ok, I will change soon.